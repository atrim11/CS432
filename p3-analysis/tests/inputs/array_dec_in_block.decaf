def int main() {
    int arr[5];
    arr = 5;  // Error: cannot assign int to int[]
}
