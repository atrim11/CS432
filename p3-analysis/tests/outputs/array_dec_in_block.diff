--- outputs/array_dec_in_block.txt	2024-10-24 17:39:00.819922608 -0400
+++ expected/array_dec_in_block.txt	2024-10-24 17:38:50.926422668 -0400
@@ -1,2 +1,2 @@
-Invalid: Array 'arr' on line 3 accessed without an index
-Invalid: Array 'arr' on line 1 declared in block
+Array 'arr' accessed without index on line 3
+Local variable 'arr' on line 2 cannot be an array
\ No newline at end of file
