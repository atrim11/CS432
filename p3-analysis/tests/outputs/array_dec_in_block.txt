Invalid: Array 'arr' on line 3 accessed without an index
Invalid: Array 'arr' on line 1 declared in block
