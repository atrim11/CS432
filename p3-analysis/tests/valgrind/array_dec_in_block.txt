==2760200== Memcheck, a memory error detector
==2760200== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==2760200== Using Valgrind-3.22.0 and LibVEX; rerun with -h for copyright info
==2760200== Command: ../decaf inputs/array_dec_in_block.decaf
==2760200== 
Invalid: Array 'arr' on line 3 accessed without an index
Invalid: Array 'arr' on line 1 declared in block
==2760200== 
==2760200== HEAP SUMMARY:
==2760200==     in use at exit: 0 bytes in 0 blocks
==2760200==   total heap usage: 1,783 allocs, 1,783 frees, 204,456 bytes allocated
==2760200== 
==2760200== All heap blocks were freed -- no leaks are possible
==2760200== 
==2760200== For lists of detected and suppressed errors, rerun with: -s
==2760200== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
