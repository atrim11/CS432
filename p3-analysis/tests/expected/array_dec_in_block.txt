Array 'arr' accessed without index on line 3
Local variable 'arr' on line 2 cannot be an array